### PR TITLE
Add Solaris 10 support

### PR DESCRIPTION
**Pull request by juliengk**

### CODE REVIEW

**Comment by ghoneycutt** [comment]
Hi @juliengk 
Appreciate the work, though it will need to be rebased since merging @Phil-Friderici 's changes for EL7 and simplifying testing.

Since this is such a huge change, please have @Phil-Friderici check the code for correctness and being complete with testing and see if someone else can confirm they got it working on Solaris 10.

**Comment by Phil-Friderici** [comment]
@juliengk please rebase and I will happily check the code and spec tests :)

**Comment by juliengk** [comment]
@Phil-Friderici rebase has been done. Thank you

**Comment by Phil-Friderici** [comment]
Review started, @juliengk expect some minor recommendations tomorrow

**Comment by Phil-Friderici** [comment]
@juliengk send a [PR](https://github.com/juliengk/puppet-module-nscd/pull/1) which adds some more negative tests. Also did some refactoring to enhance readability.

Garrett merged some README simplifications lately. Please pull, rebase and adjust your documentation.

**Comment by Phil-Friderici** [comment]
What a massive addition ! @juliengk good work !!

**Comment by juliengk** [comment]
@Phil-Friderici Let me know if anything else need to be done. Thank you

File: README.md [hunk 1]
```diff
@@ -181,207 +182,327 @@ Switch to show netgroup section in nscd.conf. Allows for boolean, 'true', or 'fa
...
+-------------------------
 Settings for enable_opt_auto_propagate in nscd.conf. Allows for boolean, 'true', or 'false'.
 
 - *Default*: 'USE_DEFAULTS'
 
 passwd_enable_cache
-----------------------
-Settings for enable-cache passwd in nscd.conf. Must be 'yes' or 'no'.
+-------------------
+Settings for enable-cache service in nscd.conf where service can be either passwd, group, hosts, services, netgroup, audit_user, auth_attr, bootparams, ethers, exec_attr, ipnodes, netmasks, networks, printers, prof_attr, project, protocols, rpc, tnrhdb, tnrhtp, user_attr. Must be 'yes' or 'no'.
```
Comment (Phil-Friderici) [inline README.md]: Please don't bring back the old bad readable documentation with this PR.
Same is true for the following parts.

File: README.md [hunk 2]
```diff
@@ -181,207 +182,327 @@ Switch to show netgroup section in nscd.conf. Allows for boolean, 'true', or 'fa
...
--------------------------------
-Settings for negative-time-to-live hosts in nscd.conf. Must be a number in seconds.
+----------------------------
+Settings for negative-time-to-live service in nscd.conf where service can be either passwd, group, hosts, services, netgroup, audit_user, auth_attr, bootparams, ethers, exec_attr, ipnodes, netmasks, networks, printers, prof_attr, project, protocols, rpc, tnrhdb, tnrhtp, user_attr. Must be a number in seconds.
 
 - *Default*: 20
 
+hosts_keep_hot_count
+---------------------
+Settings for keep-hot-count service in nscd.conf where service can be either passwd, group, hosts, services, netgroup, audit_user, auth_attr, bootparams, ethers, exec_attr, ipnodes, netmasks, networks, printers, prof_attr, project, protocols, rpc, tnrhdb, tnrhtp, user_attr. Must be a number.
```
Comment (Phil-Friderici) [inline README.md]: Please mimic the new short style for the new parameters.

File: manifests/init.pp [hunk 3]
```diff
@@ -222,6 +559,150 @@
     }
   }
 
+  if type($enable_db_audit_user) == 'boolean' {
```
Comment (ghoneycutt) [inline manifests/init.pp]: `type` is now a reserved word. Instead use `is_string()` https://github.com/puppetlabs/puppetlabs-stdlib/tree/3.2.0#is_string

Comment (juliengk) [inline manifests/init.pp]: can i do that in a second PR?

**Comment by juliengk** [comment]
@Phil-Friderici thx! README is fixed :)

**Comment by Phil-Friderici** [comment]
@juliengk sorry, I don't see your changes at the moment

**Comment by juliengk** [comment]
@Phil-Friderici forgot to push. should be good now :)

**Comment by Phil-Friderici** [comment]
@juliengk there are still 'old-style' descriptions in your commit

File: manifests/init.pp (hunk 3 above)
Comment (juliengk) [inline manifests/init.pp]: is that ok?

Comment (ghoneycutt) [inline manifests/init.pp]: You don't need a separate PR to make changes to this one. Just push your changes to this branch and they will show up here.

**Comment by juliengk** [comment]
10 out of 12 test succeed on Travis. The reason for the 2 last tests to failed is:

Cloning into 'spec/fixtures/modules/stdlib'...
fatal: read error: Connection reset by peer
rake aborted!
Failed to clone git repository git://github.com/puppetlabs/puppetlabs-stdlib.git into spec/fixtures/modules/stdlib

**Comment by ghoneycutt** [comment]
as per @Phil-Friderici comments, the README still needs work.

**Comment by juliengk** [comment]
can you point me to a direction? I don't see what you mean. Thank you

File: README.md [hunk 4]
```diff
@@ -164,242 +165,362 @@ Setting for restart-interval in nscd.conf.  See nscd.conf(5). Must be a number.
...
 
 - *Default*: 'USE_DEFAULTS'
 
 passwd_enable_cache
-----------------------
+-------------------
 Settings for enable-cache passwd in nscd.conf. Must be 'yes' or 'no'.
 
-- *Default*: 'no'
+- *Default*: 'yes'
```
Comment (Phil-Friderici) [inline README.md]: good find, thanks for fixing it !

File: README.md [hunk 5]
```diff
@@ -164,242 +165,362 @@ Setting for restart-interval in nscd.conf.  See nscd.conf(5). Must be a number.
...
 
 - *Default*: 'yes'
 
 group_enable_cache
-----------------------
+------------------
 Settings for enable-cache group in nscd.conf. Must be 'yes' or 'no'.
 
-- *Default*: 'no'
+- *Default*: 'yes'
```
Comment (Phil-Friderici) [inline README.md]: +1

File: README.md [hunk 6]
```diff
@@ -164,242 +165,362 @@ Setting for restart-interval in nscd.conf.  See nscd.conf(5). Must be a number.
...
+---------------
 Settings for shared services in nscd.conf. Must be 'yes' or 'no'.
 
 - *Default*: 'yes'
 
 services_max_db_size
----------------------
-Settings for max-db-size services in nscd.conf. Must be a number in bytes.
+--------------------
+Settings for max-db-size servicess in nscd.conf. Must be a number in bytes.
```
Comment (Phil-Friderici) [inline README.md]: services is better

**Comment by Phil-Friderici** [comment]
@juliengk only by accident I realized that you have worked on the README. Now it looks good :)
@ghoneycutt +1 from my side

**Comment by ghoneycutt** [comment]
@juliengk I'm good with this now that @Phil-Friderici has approved it.

It cannot be merged though. Could you please rebase against master?

**Comment by ghoneycutt** [comment]
@Phil-Friderici This still needs to be rebased before I can merge it

**Comment by ghoneycutt** [comment]
Aha.. I see you are doing that in #36

**Pull request closed without merging**